### PR TITLE
Refactor more LDAP code into separate package

### PR DESCRIPTION
**Pull request by alistanis**
This PR moves a few more things LDAP related into the new auth/windows package.

I don't think that the mutex from the windows desktop service conflicts with these changes, but I'm not 100% sure if I've broken consistency here, so I'd appreciate some eyes on that in particular.

This PR is gated by https://github.com/gravitational/teleport/pull/18299, which needs to go in first.

### CODE REVIEW

File: lib/srv/db/sqlserver/kinit/kinit.go
Comment (zmb3) [inline lib/srv/db/sqlserver/kinit/kinit.go:97]: Use `os.MkdirTemp` instead?

File: lib/srv/db/sqlserver/kinit/kinit.go
Comment (zmb3) [inline lib/srv/db/sqlserver/kinit/kinit.go:146]: Are you okay with leaving files behind in the temp dir if one of these writes failed? I would try to clean them up.

File: lib/srv/db/sqlserver/kinit/kinit.go
Comment (zmb3) [inline lib/srv/db/sqlserver/kinit/kinit.go:180]: Would `trace.AccessDenied` be appropriate here?

File: lib/srv/db/sqlserver/kinit/kinit.go
Comment (zmb3) [inline lib/srv/db/sqlserver/kinit/kinit.go:135]: We should probably update the trace package to make this easier. Maybe add something like `trace.New()`..

File: lib/srv/db/sqlserver/kinit/kinit.go
Comment (zmb3) [inline lib/srv/db/sqlserver/kinit/kinit.go:194]: Totally up to you, but sometimes when things like this start to get too big I prefer the use of `text/template` instead of `Sprintf` because it's easier to see which `%s` goes where.

This particular example is not too bad, so I don't feel strongly - I'll leave it to you to decide. It would look something like this when done:

```
[libdefaults]
default_realm = {{ .RealmName }}
rdns = false

[realms]
{{ .RealmName }} = {
  kdc = {{ .KDCHostName }}
  ...
```

File: lib/srv/desktop/windows_server.go
Comment (zmb3) [inline lib/srv/desktop/windows_server.go:98]: I don't think you need this at all anymore - it was only to protect `ldapInitialized` and `ldapCertRenew`.

**Comment by zmb3** [comment]
@ibeckermayer would you mind taking this for a test drive before we merge it just so we don't break desktop access again? 😄

File: lib/srv/desktop/windows_server.go
Comment (alistanis) [reply lib/srv/desktop/windows_server.go:98]: This was my biggest question here - it didn't look like it was used for anything else but I wasn't 100% sure. Thanks!

**Review by ibeckermayer — COMMENTED** [review]
It's working for me in the present state, tag me once you've addressed feedback and I'll give it another run just to confirm.

**Comment by alistanis** [comment]
> It's working for me in the present state, tag me once you've addressed feedback and I'll give it another run just to confirm.

Thanks @ibeckermayer, I appreciate it. I need to get the other PR ahead of this one in first, then I'll rebase and address this one and give you a shout.

File: lib/auth/windows/windows.go
Comment (smallinsky) [inline lib/auth/windows/windows.go:111]: Seems to be unused.

File: api/types/database.go
Comment (smallinsky) [inline api/types/database.go:645]: it will return "etc/krb5.conf" that not looks valid if process dir is not '/'

File: lib/srv/db/server.go
Comment (smallinsky) [inline lib/srv/db/server.go:76]: Where this  `LDAPConfig`  is set for the db.Config  ?

File: lib/srv/db/server.go
Comment (smallinsky) [inline lib/srv/db/server.go:442]: Why do we want always to start SQLServerLDAP ? For the old configuration when the `database.GetAD().LDAPCert` is empty the `startSQLServerLDAP` will returns an error and block the service from running.

File: lib/srv/db/server.go
Comment (smallinsky) [inline lib/srv/db/server.go:368]: Does it make sense to use domains as a key  instead of database name  and detect the situation when the initializeLDAP is already running for particular domain ?

File: lib/srv/db/server.go
Comment (smallinsky) [inline lib/srv/db/server.go:440]: @alistanis 
The startSQLServerLDAP functions calls
`startSQLServerLDAP-> InitializeLDAP-> tlsConfigForLDAP-> GenerateCredentials` 

where [GenerateCredentials](https://github.com/gravitational/teleport/blob/6d74ef49fdf079c33c7cd1e0f9a3eb5c354f9009/lib/auth/windows/windows.go#L115)
does the `authClient.GenerateWindowsDesktopCert` call 

How does the DB service `startSQLServerLDAP` call works if `GenerateWindowsDesktopCert` endpoint allows to be invoked only by `RoleWindowsDesktop` role [link](https://github.com/gravitational/teleport/blob/dd67a49ab5035e51847a3fc8934f53b5abe25a6a/lib/auth/auth_with_roles.go#L5060) ?

**Review by smallinsky — CHANGES_REQUESTED** [review]

**Comment by zmb3** [comment]
Closing due to inactivity

**Pull request closed without merging**
